src/main/java/com/example/hogwarts/services/FacultyService.java

```diff
@@ -0,0 +1,62 @@
+package com.example.hogwarts.services;
+
+import com.example.hogwarts.exceptions.FacultyNotFoundException;
+import com.example.hogwarts.model.Faculty;
+import com.example.hogwarts.repository.FacultyRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Service
+public class FacultyService {
+
+    private final FacultyRepository facultyRepository;
+
+    @Autowired
+    public FacultyService(FacultyRepository facultyRepository) {
+        this.facultyRepository = facultyRepository;
+    }
+
+    public Faculty create(Faculty faculty) {
+        return facultyRepository.save(faculty);
+    }
+
+    public Faculty get(long id) {
+        if (facultyRepository.findById(id).isPresent()) {
+            return facultyRepository.findById(id).get();
+        }
+        throw new FacultyNotFoundException(id);
```

Comment:
2 раза запрос в БД идёт, можно в одну строку
```suggestion
        return facultyRepository.findById(id).orElseThrow(() -> new FacultyNotFoundException(id));
        }
```